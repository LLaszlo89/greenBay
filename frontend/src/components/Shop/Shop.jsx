import React , { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import CardMedia from "@material-ui/core/CardMedia";
import Content from "../Content/Content";
import { findItem } from "../../redux/actions/itemsActions";
import { connect } from "react-redux";
import {CANCEL_ERR_MESSAGE} from '../../redux/actions/actionTypes'


const useStyles = makeStyles((theme) => ({
  root: {
    "& .MuiTextField-root": {
      margin: theme.spacing(1),
      width: "25ch",
      color: "#52b202",
    },
  },
  button: {
    color: "#52b202",
    fontFamily: "georgia",
  },
}));

const SaleItems = (props) => {
  const [values, setValues] = useState("");
  const [open, setOpen] = useState(false);

  const handleLoginToggle = () => {
    setOpen(!open);
    {props.notFound.message && props.cancel()}
  };

  
  const handleInputChange = (e) => {
    e.preventDefault();
    const { value } = e.target;
    setValues(value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    props.searchId(values);
    handleLoginToggle();
  };

  const classes = useStyles();

  const sellerStatus =  props.specId.soldTo ? "Sold to :" + props.specId.soldTo : " Still Available ! "
  const status = props.notFound  ? (
    <div>
      <Grid container direction="row" justify="center" alignContent="center" >
        <Grid>
          <Typography align="center" variant="h6" color="error">
            {props.notFound.message}
          </Typography>
        </Grid>
      </Grid>
    </div>
  ) : (
    <div>
      <Grid
        container
        direction="row"
        justify="center"
        alignContent="center"
      >
        <Grid>
          <CardMedia
            component="img"
            style={{ maxWidth: "150px", borderRadius: "10px" }}
            image={props.specId.URL}
          />
        </Grid>
        <Grid>
          <Typography align="center" variant="h4">
            {"Title :" + props.specId.title}
          </Typography>
        </Grid>
      </Grid>
      <Typography align="center" variant="h6">
        {`ID number : ${props.specId.id} , SELLER :  ${props.specId.user_name} `}
      </Typography>
      <Typography align="center">
        {"DESCRIPTION :" + props.specId.description}
      </Typography>
      <Typography style={{ marginRight:"25px"}} align="right">{`PRICE : ${props.specId.price} $, ${sellerStatus}`}</Typography>
    </div>
  );
console.log("This is the mssg in shop after buy" , props.purchase_message)
  return (
    <div>
      <form className={classes.root} noValidate autoComplete="off">
        <div>
          <Grid container direction="row" alignContent="center">
            <TextField
              placeholder="Enter Id "
              id="outlined-search"
              label="Search item"
              type="search"
              name="search"
              variant="outlined"
              onChange={handleInputChange}
            />
            <Button
              className={classes.button}
              type="submit"
              onClick={handleSubmit}
              variant="outlined"
            >
              Search
            </Button>
          </Grid>
            <Typography align="center" variant="h6">
            {props.purchase_message}
          </Typography>
        </div>
      </form>
      <Dialog open={open} onClose={handleLoginToggle} fullWidth>
        <DialogTitle id="form-dialog-title">
          Information about the requested item:{" "}
        </DialogTitle>
        {status}
      </Dialog>
      <Content />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    specId: state.items.spec_id,
    notFound: state.items.err_message,
    purchase_message: state.items.sold_message
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    searchId: (id) => dispatch(findItem(id)),
    cancel: () => dispatch({ type : CANCEL_ERR_MESSAGE })
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SaleItems);
