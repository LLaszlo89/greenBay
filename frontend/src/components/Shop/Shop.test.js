import React from "react";
import renderer from "react-test-renderer";
import Shop from "./Shop";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Connect component to redux store ", () => {
  const mockStore = configureStore([]);

  const items = {
    items: [
      {
        id: 1,
        user_name: "Test",
        title: "Test",
        description: "Test",
        URL: "Test",
        price: 1,
        forSale: 1,
      },
    ],
    spec_id :"",
    err_message:"",
    sold_message:""
  };


  const store = mockStore({ items });

  it("Shop Component should render correctly", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Shop />
        </Provider>
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
