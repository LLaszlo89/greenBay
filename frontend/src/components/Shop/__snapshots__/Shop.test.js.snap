// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to redux store  Shop Component should render correctly 1`] = `
<div>
  <form
    autoComplete="off"
    className="makeStyles-root-1"
    noValidate={true}
  >
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-content-xs-center"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-search"
            id="outlined-search-label"
          >
            Search item
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
              disabled={false}
              id="outlined-search"
              name="search"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter Id "
              required={false}
              type="search"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-5"
              >
                <span>
                  Search item
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Search
          </span>
        </button>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
      >
        
      </h6>
    </div>
  </form>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Test
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                1 green_$
              </span>
            </div>
          </div>
          <div
            className="MuiCardMedia-root makeStyles-img-8"
            style={
              Object {
                "backgroundImage": "url(\\"Test\\")",
              }
            }
          />
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-9 MuiButton-textSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                BUY NOW
              </span>
            </button>
          </div>
          <div
            className="MuiCardContent-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Test
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
